addons/osio-addon/src/main/java/io/fabric8/launcher/osio/preparers/InsertMavenDependenciesPreparer.java

```diff
@@ -0,0 +1,54 @@
+package io.fabric8.launcher.osio.preparers;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.fabric8.launcher.base.maven.Maven;
+import io.fabric8.launcher.booster.catalog.rhoar.RhoarBooster;
+import io.fabric8.launcher.core.api.ProjectileContext;
+import io.fabric8.launcher.core.spi.ProjectilePreparer;
+import io.fabric8.launcher.osio.projectiles.context.OsioProjectileContext;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Dependency;
+
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class InsertMavenDependenciesPreparer implements ProjectilePreparer {
+
+    private final Logger LOG = Logger.getLogger(InsertMavenDependenciesPreparer.class.getName());
+
+
+    @Override
+    public void prepare(Path projectPath, RhoarBooster booster, ProjectileContext context) {
+        if (!(context instanceof OsioProjectileContext)) {
+            return;
+        }
+        OsioProjectileContext createProjectileContext = (OsioProjectileContext) context;
+        Path pom = projectPath.resolve("pom.xml");
+        List<String> dependencies = createProjectileContext.getDependencies();
+
+        // Perform model changes
+        if (Files.isRegularFile(pom) && !dependencies.isEmpty()) {
+            try {
+                Model model = Maven.readModel(pom);
+                // Get the list of dependencies in the format of 'groupId:artifactId:version'
+                for (String dep : dependencies) {
+                    Dependency dependency = new Dependency();
+                    String[] groupIdArtifactIdVersion = dep.split(":");
+                    dependency.setGroupId(groupIdArtifactIdVersion[0]);
+                    dependency.setArtifactId(groupIdArtifactIdVersion[1]);
+                    dependency.setVersion(groupIdArtifactIdVersion[2]);
+                    model.addDependency(dependency);
```

Comment:
What if the dependency already exists? Maven will throw an error if duplicate dependencies are found in the pom.xml

Reply:
@gastaldi what should be the expected behavior if a user wants to add a dependency which already exists but the version is different.

Reply:
Use the version that is in the list of dependencies in the ProjectileContext

Reply:
ok, Thanks